### PR TITLE
feat(expect): add `expect.{closeTo, stringContaining, stringMatching}`

### PR DESCRIPTION
**Pull request by babiabeo**
This PR adds these following asymmetric matchers:

- [x] expect.closedTo
- [x] expect.stringContaining
- [x] expect.not.stringContaining
- [x] expect.stringMatching
- [x] expect.not.stringMatching

Ref: #3964

### CODE REVIEW

File: expect/mod.ts [hunk 1]
```diff
@@ -48,25 +48,29 @@
...
+ *   - `expect.objectContaining`
+ *   - `expect.not.objectContaining`
  *
- * The following uitlities are not supported yet:
- * - `expect.assertions`
- * - `expect.hasAssertions`
- * - `expect.addEqualityTester`
- * - `expect.addSnapshotSerializer`
- * - `expect.extend`
+ * The uitlities are not yet fully supported:
```
Comment (iuioiua) [inline expect/mod.ts:66]:
```suggestion
 * The utilities are not yet fully supported:
```

File: expect/mod.ts [hunk 2]
```diff
@@ -48,25 +48,29 @@
...
+ *   - `expect.arrayContaining`
+ *   - `expect.not.arrayContaining`
+ *   - `expect.closedTo`
+ *   - `expect.stringContaining`
+ *   - `expect.not.stringContaining`
+ *   - `expect.stringMatching`
+ *   - `expect.not.stringMatching`
+ * - Not supported:
+ *   - `expect.objectContaining`
+ *   - `expect.not.objectContaining`
```
Comment (iuioiua) [inline expect/mod.ts:64]: This doesn't make sense. Within a list of symbols that are not fully supported, there's a list of symbols that are supported. Can you please correct and clarify?

File: expect/mod.ts [hunk 3]
```diff
@@ -48,25 +48,29 @@
...
- * - `expect.addSnapshotSerializer`
- * - `expect.extend`
+ * The uitlities are not yet fully supported:
+ * - Supported:
+ *   - `expect.addEqualityTester`
+ *   - `expect.extend`
+ * - Not supported:
+ *   - `expect.assertions`
+ *   - `expect.hasAssertions`
+ *   - `expect.addSnapshotSerializer`
```
Comment (iuioiua) [inline expect/mod.ts:73]: Ditto. This doesn't make sense for the same reason as the above list. Please correct and clarify.

File: expect/_asymmetric_matchers.ts [hunk 4]
```diff
@@ -69,11 +70,78 @@ export class ArrayContaining extends AsymmetricMatcher<any[]> {
...
+  equals(other: string): boolean {
+    if (typeof other !== "string") {
+      return false;
+    }
+
+    return other.includes(this.value);
+  }
+}
+
+export function stringContaining(str: string) {
```
Comment (iuioiua) [inline expect/_asymmetric_matchers.ts:127]: Please don't forget to define explicit return types (except for `void` or `Promise<void>`). Ditto elsewhere.

**Review by iuioiua — CHANGES_REQUESTED** [review]

**Comment by babiabeo** [comment]
Let me know if these changes make sense to you. 👍

**Review by kt3k — APPROVED** [review]
LGTM. Nice work!

**Pull request merged**